Classes/WPEditorToolbarButton.m

```diff
@@ -170,21 +167,29 @@ - (void)setSelected:(BOOL)selected
 	[super setSelected:selected];
 	
 	if (hasChangedSelectedStatus) {
-		dispatch_time_t dispatchDelay = dispatch_time(DISPATCH_TIME_NOW,
-													  (int64_t)(0.2 * NSEC_PER_SEC));
-		
-		dispatch_after(dispatchDelay, dispatch_get_main_queue(), ^{
-			if (selected) {
-				self.tintColor = self.selectedTintColor;
-				[self slideInBottomLineView];
-			} else {
-				self.tintColor = self.normalTintColor;
-				[self slideOutBottomLineView];
-			}
-		});
+        if (self.enabled) {
+            if (selected) {
+                self.tintColor = self.selectedTintColor;
+            } else {
+                self.tintColor = self.normalTintColor;
+            }
+        } else {
+            self.tintColor = self.disabledTintColor;
+        }
 	}
 }
 
+- (void)setEnabled:(BOOL)enabled
+{
+    [super setEnabled:enabled];
+    
+    if (enabled) {
+        self.tintColor = self.normalTintColor;
```

Comment:
Should this also check for selected status before assigning the `normalTintColor`?